streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java

```diff
@@ -25,18 +25,57 @@
 public interface StreamsMetrics {
 
     /**
-     * Add the latency sensor.
+     * Add a latency sensor. This is equivalent to adding a sensor with metrics on latency and rate.
      *
      * @param scopeName Name of the scope, could be the type of the state store, etc.
      * @param entityName Name of the entity, could be the name of the state store instance, etc.
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
      * @param operationName Name of the operation, could be get / put / delete / etc.
      * @param tags Additional tags of the sensor.
      * @return The added sensor.
      */
-    Sensor addLatencySensor(String scopeName, String entityName, String operationName, String... tags);
+    Sensor addLatencySensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags);
 
     /**
      * Record the given latency value of the sensor.
      */
     void recordLatency(Sensor sensor, long startNs, long endNs);
+
+    /**
+     * Add a throughput sensor. This is equivalent to adding a sensor with metrics rate.
+     *
+     * @param scopeName Name of the scope, could be the type of the state store, etc.
+     * @param entityName Name of the entity, could be the name of the state store instance, etc.
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
+     * @param operationName Name of the operation, could be get / put / delete / etc.
+     * @param tags Additional tags of the sensor.
+     * @return The added sensor.
+     */
+    Sensor addThroughputSensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags);
+
+
+    /**
+     * Generic sensor creation. Note that for most cases it is advisable to use {@link #addThroughputSensor(String, String, String, Sensor.RecordLevel, String...)}
+     * of {@link #addLatencySensor(String, String, String, Sensor.RecordLevel, String...)} to ensure metric name well-formedness and conformity with the rest
+     * of the streams code base.
+     * @param scopeName Name of the scope, could be the type of the state store, etc.
+     * @param entityName Name of the entity, could be the name of the state store instance, etc.
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
+     * @param operationName Name of the operation, could be get / put / delete / etc.
+     */
+    Sensor sensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel);
```

Comment:
The answer is that we'll provide helper functions but also expose the metrics registry as well.